static/js/hotkey.js

```diff
@@ -44,6 +46,41 @@ function do_narrow_action(action) {
     return true;
 }
 
+function handle_editing_last_sent_message() {
+    // It is necessary to check if there are unreads in the
+    // given view before editing the last sent message in the view.
+    // If the unreads in the given view are older than
+    // the last sent message in the view, using the hotkey to directly
+    // edit the last sent message will mark the unreads as read.
+
+    const last_user_msg = message_lists.current.get_last_message_sent_by_me();
+    const first_unread_msg_id = message_lists.current.first_unread_message_id();
+    let has_unreads = false;
+    if (last_user_msg && first_unread_msg_id) {
+        const first_unread_msg = message_store.get(first_unread_msg_id);
+        if (last_user_msg.id > first_unread_msg.id) {
+            has_unreads = true;
+        }
+    }
+
+    if (!has_unreads) {
+        message_edit.edit_last_sent_message();
+    } else {
+        feedback_widget.show({
+            populate(container) {
+                container.html(
+                    $t({defaultMessage: "Editing last sent message may mark unreads as read."}),
```

Comment:
We should be more clear here: "Editing the last message you sent will mark all messages in this view as read" -- since I think that's true?

Reply:
Yeah, that's right. Will update.